tests/lib/common/github/issuereporter.ts

```diff
@@ -0,0 +1,212 @@
+import anyTest, { TestInterface } from 'ava';
+import * as moment from 'moment';
+import * as sinon from 'sinon';
+import * as proxyquire from 'proxyquire';
+
+type OctokitSearch = {
+    issues: () => Promise<any>;
+};
+
+type IssueReporterContext = {
+    Octokit: () => void;
+    sandbox: sinon.SinonSandbox;
+}
+
+const test = anyTest as TestInterface<IssueReporterContext>;
+
+const loadScript = (context: IssueReporterContext) => {
+    return proxyquire('../../../../src/lib/common/github/issuereporter', { '@octokit/rest': context.Octokit }).IssueReporter;
+};
+
+test.beforeEach((t) => {
+    t.context.sandbox = sinon.createSandbox();
+
+    const Octokit = function () {
+    };
+
+    Octokit.prototype.authenticate = () => { };
+    Octokit.prototype.issues = {
+        create() { },
+        createComment() { },
+        update() { }
+    };
+    Octokit.prototype.search = {
+        issues(): Promise<any> {
+            return null;
+        }
+    };
+
+    t.context.Octokit = Octokit;
+});
+
+test('If no error and no issue, nothing happens', async (t) => {
+    const sandbox = t.context.sandbox;
+    const Octokit = t.context.Octokit;
+
+    const octokitSearchIssuesStub = sandbox.stub<OctokitSearch, 'issues'>(Octokit.prototype.search, 'issues').resolves({ data: { items: [] } });
+    const octokitIssuesUpdateSpy = sandbox.spy(Octokit.prototype.issues, 'update');
+    const octokitIssuesCreateSpy = sandbox.spy(Octokit.prototype.issues, 'create');
+    const IssueReporter = loadScript(t.context);
+    const issueReporter = new IssueReporter();
+
+    await issueReporter.report({ configs: [], scan: moment().format('YYYY-MM-DD'), url: 'http://example.com' });
+
+    t.true(octokitSearchIssuesStub.called);
+    t.false(octokitIssuesUpdateSpy.called);
+    t.false(octokitIssuesCreateSpy.called);
+
+    sandbox.restore();
+});
+
+test('If no error but issue exists, it should close the issue', async (t) => {
+    const sandbox = t.context.sandbox;
+    const Octokit = t.context.Octokit;
+
+    const octokitSearchIssuesStub = sandbox.stub<OctokitSearch, 'issues'>(Octokit.prototype.search, 'issues').resolves({ data: { items: [{ issue_number: 1 }] } }); // eslint-disable-line camelcase
+    const octokitIssuesUpdateStub = sandbox.stub(Octokit.prototype.issues, 'update').resolves();
+    const octokitIssuesCreateSpy = sandbox.spy(Octokit.prototype.issues, 'create');
+    const IssueReporter = loadScript(t.context);
+    const issueReporter = new IssueReporter();
+
+    await issueReporter.report({ configs: [], scan: moment().format('YYYY-MM-DD'), url: 'http://example.com' });
+
+    t.true(octokitSearchIssuesStub.called);
+    t.true(octokitIssuesUpdateStub.calledOnce);
+    t.false(octokitIssuesCreateSpy.called);
+
+    const args = octokitIssuesUpdateStub.args[0][0];
+
+    t.is(args.state, 'closed');
+    t.is(args.number, 1);
```

Comment:
number should not exist
```suggestion
    t.is(args.issue_number, 1);
```